controllers/renovate_util.go

```diff
@@ -22,63 +24,173 @@ import (
 	"github.com/konflux-ci/build-service/pkg/logs"
 )
 
 const (
 	RenovateImageEnvName    = "RENOVATE_IMAGE"
 	DefaultRenovateImageUrl = "quay.io/redhat-appstudio/renovate:v37.74.1"
+	DefaultRenovateUser     = "redhat-hat-konflux"
 )
 
-type installationStruct struct {
-	id           int
-	component    string
-	token        string
-	repositories []renovateRepository
-}
-
 type renovateRepository struct {
 	Repository   string   `json:"repository"`
 	BaseBranches []string `json:"baseBranches,omitempty"`
 }
 
-// GetGithubInstallationsForComponents This method avoids iterating over all installations, it is intended to be called when the component list is small
-func GetGithubInstallationsForComponents(ctx context.Context, client client.Client, eventRecorder record.EventRecorder, componentList []v1alpha1.Component) (string, []installationStruct, error) {
+// UpdateTarget represents a target source code repository to be executed by Renovate with credentials and repositories
+type updateTarget struct {
+	ComponentName string
+	GitProvider   string
+	Username      string
+	GitAuthor     string
+	Token         string
+	Endpoint      string
+	Repositories  []renovateRepository
+}
+
+/*
+type ComponentDependenciesUpdater interface {
+	GetUpdateTargetsGithubApp(ctx context.Context, componentList []v1alpha1.Component) []updateTarget
+	GetUpdateTargetsBasicAuth(ctx context.Context, componentList []v1alpha1.Component) []updateTarget
+	GenerateRenovateConfigForNudge(target updateTarget, buildResult *BuildResult) (string, error)
+	CreateRenovaterPipeline(ctx context.Context, namespace string, targets []updateTarget, debug bool, buildResult *BuildResult) error
+}
+*/
+
+type DefaultComponentDependenciesUpdater struct {
+	Client             client.Client
+	Scheme             *runtime.Scheme
+	EventRecorder      record.EventRecorder
+	CredentialProvider *k8s.GitCredentialProvider
+}
+
+func NewDefaultComponentDependenciesUpdater(client client.Client, scheme *runtime.Scheme, eventRecorder record.EventRecorder) *DefaultComponentDependenciesUpdater {
+	return &DefaultComponentDependenciesUpdater{Client: client, Scheme: scheme, EventRecorder: eventRecorder, CredentialProvider: k8s.NewGitCredentialProvider(client)}
+}
+
+// GetUpdateTargetsBasicAuth This method returns targets for components based on basic auth
+func (d DefaultComponentDependenciesUpdater) GetUpdateTargetsBasicAuth(ctx context.Context, componentList []v1alpha1.Component) []updateTarget {
+	log := logger.FromContext(ctx)
+	targetsToUpdate := []updateTarget{}
+
+	for _, component := range componentList {
+		gitProvider, err := getGitProvider(component)
+		if err != nil {
+			log.Error(err, "error detecting git provider", "ComponentName", component.Name, "ComponentNamespace", component.Namespace)
+			continue
+		}
+
+		scmComponent, err := git.NewScmComponent(gitProvider, component.Spec.Source.GitSource.URL, component.Spec.Source.GitSource.Revision, component.Name, component.Namespace)
```

Comment:
After removing of renovate for tekton updates, this `ScmComponent` looks redundant... Why not just pass Component?

Reply:
GetBasicAuthCredentials needs scmComponent

also I am using that url parsing to get host

Reply:
I'm not sure that keeping an abstraction just for GetBasicAuthCredentials and url parsing is a good idea, but up to you.